distributed/shuffle/tests/test_shuffle.py

```diff
@@ -300,12 +301,39 @@ async def test_closed_worker_during_transfer(c, s, a, b):
         freq="10 s",
     )
     out = dd.shuffle.shuffle(df, "x", shuffle="p2p")
-    out = out.persist()
+    x, y = c.compute([df.x.size, out.x.size])
     await wait_for_tasks_in_state("shuffle-transfer", "memory", 1, b)
     await b.close()
 
-    with pytest.raises(RuntimeError):
-        out = await c.compute(out)
+    x = await x
+    y = await y
+    assert x == y
+
+    await c.close()
+    await check_worker_cleanup(a)
+    await check_worker_cleanup(b, closed=True)
+    await check_scheduler_cleanup(s)
+
+
+@gen_cluster(
+    client=True,
+    nthreads=[("", 1)] * 2,
+    config={"distributed.scheduler.allowed-failures": 0},
```

Comment:
does this mean that P2P is now retried `allowed-failures` times if a worker goes OOM?

Reply:
Wouldn't be a dealbreaker but I also don't think this is useful. It's very unlikely that another P2P run attempt would be more successful.

However, there are of course also cases like spot interruption where this matters... Never mind!

Reply:
> does this mean that P2P is now retried allowed-failures times if a worker goes OOM?
Yes, as there might be other causes apart from an output partition being too large.